src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -271,49 +596,130 @@ def __init__(
+                )
 
-        #########
-        # Extract and re-name properties from nwbfile TODO: Should be a function
-        ########
+        # Open the electrical series. In case of failure, raise an error with the available options.
+        try:
+            electrical_series = open_file[self.electrical_series_path]
+        except KeyError:
+            available_electrical_series = find_neurodata_type_from_backend(
+                open_file, neurodata_type="ElectricalSeries", backend=self.backend
+            )
+            raise ValueError(
+                f"{self.electrical_series_path} not found in the NWB file!"
+                f"Available options are: {available_electrical_series}."
+            )
+        electrodes_indices = retrieve_electrodes_indices_from_electrical_series(
+            open_file, electrical_series, self.backend
+        )
+        # The table for all the electrodes in the nwbfile
+        electrodes_table = open_file["/general/extracellular_ephys/electrodes"]
+        electrode_table_columns = electrodes_table.attrs["colnames"]
 
-        properties = dict()
-        # Extract rel_x, rel_y and rel_z and assign to location
+        # Get sampling frequency
+        if "starting_time" in electrical_series.keys():
+            t_start = electrical_series["starting_time"][()]
+            sampling_frequency = electrical_series["starting_time"].attrs["rate"]
+        elif "timestamps" in electrical_series.keys():
+            timestamps = electrical_series["timestamps"][:]
+            t_start = timestamps[0]
+            sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
 
-        # TODO: Refactor ALL of this and add tests. This is difficult to read.
-        if "rel_x" in electrodes_table:
-            ndim = 3 if "rel_z" in electrodes_table else 2
-            properties["location"] = np.zeros((self.get_num_channels(), ndim), dtype=float)
+        if load_time_vector and timestamps is not None:
+            times_kwargs = dict(time_vector=electrical_series.timestamps)
+        else:
+            times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
+
+        # If channel names are present, use them as channel_ids instead of the electrode ids
+        if "channel_name" in electrode_table_columns:
+            channel_names = electrodes_table["channel_name"]
+            channel_ids = channel_names[electrodes_indices]
+            # Decode if bytes with utf-8
+            channel_ids = [x.decode("utf-8") if isinstance(x, bytes) else x for x in channel_ids]
+
+        else:
+            channel_ids = [electrodes_table["id"][x] for x in electrodes_indices]
+
+        dtype = electrical_series["data"].dtype
+        electrical_series_data = electrical_series["data"]
+
+        # need this for later
+        self.electrical_series = electrical_series
+        self._file = open_file
```

Comment:
Good to make this a private attribute as it is only available on these path. Altought I think we could get the file from the nwbfile in the other path and exposed it as well but probably not worth the effort.